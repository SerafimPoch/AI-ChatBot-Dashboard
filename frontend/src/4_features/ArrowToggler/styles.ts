import { useThemeContext } from "@/src/6_shared/contexts/ThemeContext";
import styled from "@emotion/styled";

export const ArrowTogglerContainer = styled.div`
  ${() => {
    const themeColors = useThemeContext();

    return `
      display: flex;
      flex-direction: column;
      align-items: center;
      justify-content: center;
      gap: 32px;
      align-self: stretch;
      border-radius: 12px;
      background: ${themeColors?.grey};
      width: 56px;
    `;
  }}
`;
