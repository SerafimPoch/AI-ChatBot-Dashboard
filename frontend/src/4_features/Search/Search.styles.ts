import { useThemeColor } from "@/src/6_shared/hooks/useThemeColor";
import styled from "@emotion/styled";

export const SearchContainer = styled.div`
  position: relative;
  display: flex;
  align-items: center;
  gap: 8px;
`;

export const SearchInput = styled.input`
  background: transparent;
  border: 1px solid #5d74f1;
  outline: none;
  border-radius: 6px;
  padding: 0 5px;
  color: ${() => useThemeColor().textPrimary};
`;
