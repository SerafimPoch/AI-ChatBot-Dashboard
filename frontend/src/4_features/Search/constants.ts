export const SEARCH_IMAGE_AREA = { width: 48, height: 48 };
