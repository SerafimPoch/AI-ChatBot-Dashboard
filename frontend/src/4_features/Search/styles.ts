import { useThemeContext } from "@/src/6_shared/contexts/ThemeContext";
import styled from "@emotion/styled";

export const SearchContainer = styled.div`
  position: relative;
  display: flex;
  align-items: center;
  gap: 8px;
`;

export const SearchInput = styled.input`
  ${() => {
    const themeColors = useThemeContext();

    return `
      background: transparent;
      border: 1px solid #5d74f1;
      outline: none;
      border-radius: 6px;
      padding: 0 5px;
      color: ${themeColors?.textPrimary};
    `;
  }}
`;
