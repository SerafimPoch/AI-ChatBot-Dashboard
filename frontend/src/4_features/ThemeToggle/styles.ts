import styled from "@emotion/styled";

interface IThemeItem {
  active: boolean;
}

export const ThemeToggleContainer = styled.div`
  display: flex;
  align-items: center;
  border-radius: 8px;
  border: 2px solid #f2f2f2;
  width: 100%;
  height: 40px;
`;

export const ThemeItem = styled.button<IThemeItem>`
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 6px;
  width: 50%;
  height: 32px;
  background: ${({ active }) => (active ? "#f2f2f2" : "transparent")};
  color: ${({ active }) =>
    active ? "var(--text-primary)" : "var(--text-secondary)"};
  margin: 0 4px;
  transition: background-color 0.3s ease, color 0.3s ease;
`;
