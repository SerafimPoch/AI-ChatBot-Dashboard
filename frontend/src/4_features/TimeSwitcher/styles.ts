import styled from "@emotion/styled";

interface TTimeSwitcherItem {
  active: boolean;
}

export const TimeSwitcherContainer = styled.div`
  display: flex;
  align-items: center;
  gap: 16px;
`;

export const TimeSwitcherItem = styled.button<TTimeSwitcherItem>`
  width: fit-content;
  min-width: 33px;
  height: 28px;
  padding: 4px 8px;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 14px;
  font-weight: 600;
  line-height: 20px;
  font-family: var(--font-sf-pro-rounded);
  color: ${({ active }) => (active ? "#fff" : "#A0A5A9")};
  background-color: ${({ active }) => (active ? "#1D1D1D" : "transparent")};
  border-radius: 8px;
  transition: background-color 0.3s ease, color 0.3s ease;

  &:hover {
    color: #fff;
    background-color: #1d1d1d;
  }
`;
