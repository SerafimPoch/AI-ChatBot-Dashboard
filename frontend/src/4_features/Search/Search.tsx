import { useState } from "react";
import Image from "next/image";
import { SearchContainer } from "./Search.styles";
import searchHover from "./assets/search-hover.svg?url";
import searchDay from "./assets/search-day.svg?url";
import { SEARCH_IMAGE_AREA } from "./constants";

export default function Search() {
  const [isSearchOpen, setIsSearchOpen] = useState(false);

  const handleSearch = () => setIsSearchOpen(!isSearchOpen);

  return (
    <SearchContainer>
      {isSearchOpen && <input type="search" />}
      <button onClick={() => handleSearch()}>
        <Image
          src={isSearchOpen ? searchHover : searchDay}
          alt="arrow"
          width={SEARCH_IMAGE_AREA.width}
          height={SEARCH_IMAGE_AREA.height}
        />
      </button>
    </SearchContainer>
  );
}
