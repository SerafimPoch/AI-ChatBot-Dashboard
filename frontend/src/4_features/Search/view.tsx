import { useMemo, useState } from "react";
import Image from "next/image";
import { useStore } from "@nanostores/react";
import { $theme } from "@/src/6_shared/store";
import { THEME_LIGHT_KEY } from "@/src/6_shared/constants";
import { SearchContainer, SearchInput } from "./styles";
import searchDay from "./ui/assets/search-day.svg?url";
import searchNight from "./ui/assets/search-night.svg?url";

export default function Search() {
  const [isSearchOpen, setIsSearchOpen] = useState(false);

  const theme = useStore($theme);

  const searchIcon = useMemo(
    () => (theme.type === THEME_LIGHT_KEY ? searchDay : searchNight),
    [theme.type]
  );

  const handleSearch = () => setIsSearchOpen(!isSearchOpen);

  return (
    <SearchContainer>
      {isSearchOpen && (
        <SearchInput type="search" onBlur={() => handleSearch()} />
      )}
      <button onClick={() => handleSearch()}>
        <Image src={searchIcon} alt="search" width={48} height={48} />
      </button>
    </SearchContainer>
  );
}
