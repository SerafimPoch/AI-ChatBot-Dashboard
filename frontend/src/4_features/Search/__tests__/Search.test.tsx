import React from "react";
import { render } from "@testing-library/react";
import { SearchContainer, SearchInput } from "../Search.styles";

describe("Search page components", () => {
  it("SearchContainer renders correctly", () => {
    const { asFragment } = render(<SearchContainer />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("SearchInput renders correctly", () => {
    const { asFragment } = render(<SearchInput />);
    expect(asFragment()).toMatchSnapshot();
  });
});
