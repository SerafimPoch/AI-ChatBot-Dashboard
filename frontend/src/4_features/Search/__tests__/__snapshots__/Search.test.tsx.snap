// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search page components SearchContainer renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 8px;
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Search page components SearchInput renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  background: transparent;
  border: 1px solid #5d74f1;
  outline: none;
  border-radius: 6px;
  padding: 0 5px;
  color: undefined;
}

<input
    class="emotion-0"
  />
</DocumentFragment>
`;
